staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/client_test.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"testing"
+
+	"k8s.io/api/admissionregistration/v1beta1"
+)
+
+func TestHookClient(t *testing.T) {
+	scenarios := []struct {
+		clientConfig v1beta1.WebhookClientConfig
+		expectedUrl  string
+		expectError  bool
+	}{
+		// scenario_0: A valid https URL with a CABundle provided.
+		{
+			clientConfig: v1beta1.WebhookClientConfig{
+				URL:      strPtr("https://k8s.io:8080/"),
+				CABundle: []byte("123"),
+			},
+			expectedUrl: "https://k8s.io:8080/",
+		},
+		// scenario_1: A valid https URL without a CABundle provided.
+		{
+			clientConfig: v1beta1.WebhookClientConfig{
+				URL: strPtr("https://k8s.io:8080/"),
+			},
+			expectedUrl: "https://k8s.io:8080/",
+		},
+		// scenario_2: A valid http URL without a CABundle provided.
```

Comment:
The [comment](https://github.com/kubernetes/kubernetes/blob/2f4d6b863d5784b8d83c0750f7f1af9b7a9b5b98/staging/src/k8s.io/api/admissionregistration/v1beta1/types.go#L239) says that the scheme must be "https". Shall we add a check / fix the comment?